feat(deps): support latest Cerebral 0.33 and new releases

christianalfoni opened the pull request.

Guria commented:
why not to use alias instead of factory?

Guria commented:
https://github.com/cerebral/cerebral-module-recorder/blob/newContextSignals/index.js#L33
replace `isSync` with `immediate` option

Guria commented:
set [`cerebral@0.33.31`](https://github.com/cerebral/cerebral/releases/tag/v0.33.31) as minimal peerVersion

Guria commented:
port [this changes](https://github.com/cerebral/cerebral-module-recorder/pull/5/files#diff-168726dbe96b3ce427e7fedce31bb0bcR153) too

christianalfoni commented:
1. Do not have access to the alias in 0.34 inside actions (without adding a context provider for modules)
2. Changed to immediate now :)
3. Changed minimal version
4. Moved over the IF thingy there as well. Tests and deps was up to date

Guria commented on index.js:
```diff
@@ -166,6 +169,15 @@ module.exports = function (options) {
     module.alias(MODULE)
 
+    if (module.addContextProvider) {
+      controller.addContextProvider({
```
we already have MODULE const. How about:
```js
var context = {}
context[MODULE] = { path: module.path }
controller.addContextProvider(context)
```

christianalfoni replied:
agreed

Guria commented on index.js:
```diff
@@ -166,6 +169,15 @@ module.exports = function (options) {
 
     module.alias(MODULE)
 
+    if (module.addContextProvider) {
```
maybe better to check `controller.addContextProvider` here?

christianalfoni replied:
agreed

Guria commented on signals/index.js:
```diff
@@ -1,43 +1,63 @@
-  arg.module.state.merge([], {
+function play (context) {
+  var modulePath = context.modules ? context.modules['cerebral-module-recorder'].path : context['cerebral-module-recorder'].path
+  var services = modulePath.reduce(function (services, key) {
```
maybe `var services = getModuleServices(context)`?

christianalfoni replied:
agreed :)

christianalfoni commented:
Okay, implemented all your suggestions, very good :)

Guria commented:
LGTM

Guria commented:
please squash and use:
```js
"peerDependencies": {
  "cerebral": "^0.33.31 || ^0.34.0-rc.4"
},
```
